lib/buffer-search.js

```diff
@@ -224,6 +224,8 @@ class BufferSearch {
       }
 
       const newMarkers = this.createMarkers(scanStart, scanEnd);
+      if(newMarkers === false || !newMarkers.length) return; // Invalid/missing find pattern or no new markers
```

Comment:
Yeah I thought I noticed a regression related to that, but didn't realize what the steps to reproduce were.

I'll update this PR to only focus on the latter changes.